### PR TITLE
Feature/add redux structure

### PR DESCRIPTION
**Pull request by Yojuks**
http://localhost:3000/test-redux-components  - this is path for test url

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -10,14 +10,19 @@
...
     "leaflet": "^1.7.1",
     "lint-staged": "^12.3.7",
     "react": "^18.0.0",
     "react-dom": "^18.0.0",
     "react-leaflet": "4.0.0",
+    "react-redux": "^8.0.1",
     "react-router-dom": "^6.3.0",
     "react-scripts": "5.0.0",
+    "redux": "^4.2.0",
+    "redux-devtools-extension": "^2.13.9",
```
Comment (steadyGuy) [inline frontend/package.json:24]: devDependecies

File: frontend/package.json [hunk 2]
```diff
@@ -10,14 +10,19 @@
     "@testing-library/jest-dom": "^5.14.1",
     "@testing-library/react": "^12.0.0",
     "@testing-library/user-event": "^13.2.1",
+    "axios": "^0.27.2",
```
Comment (steadyGuy) [inline frontend/package.json:13]: so we decided to use axios?

File: frontend/src/redux/actions-creators/todo.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      setTimeout(() => {
+        dispatch({
+          type: TodoActionTypes.FETCH_TODOS_SUCCESS,
+          payload: response.data
+        });
+      }, 500);
+    } catch (e) {
+      dispatch({
+        type: TodoActionTypes.FETCH_TODOS_ERROR,
+        payload: 'Произошла ошибка при загрузке списка дел'
```
Comment (steadyGuy) [inline frontend/src/redux/actions-creators/todo.ts]: russian?

File: frontend/src/redux/actions-creators/user.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    setTimeout(() => {
+      dispatch({
+        type: UserActionTypes.FETCH_USERS_SUCCESS,
+        payload: response.data
+      });
+    }, 500);
+  } catch (e) {
+    dispatch({
+      type: UserActionTypes.FETCH_USERS_ERROR,
+      payload: 'Произошла ошибка при загрузке пользователей'
```
Comment (steadyGuy) [inline frontend/src/redux/actions-creators/user.ts]: maybe we need some distinct file with constants for all of the unchangeable (static) strings

File: frontend/src/redux/components/TodoList.tsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+interface Props {
+  name: 'name';
+}
+
+function TodoList(): React.ReactElement<Props> {
+  const { page, error, loading, todos, limit } = useTypedSelector(
+    state => state.todo
+  );
+  // eslint-disable-next-line no-shadow
```
Comment (steadyGuy) [inline frontend/src/redux/components/TodoList.tsx]: ?

File: frontend/src/redux/types/user.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  type: UserActionTypes.FETCH_USERS;
+}
+interface FetchUsersSuccessAction {
+  type: UserActionTypes.FETCH_USERS_SUCCESS;
+  payload: any[];
+}
+interface FetchUsersErrorAction {
+  type: UserActionTypes.FETCH_USERS_ERROR;
+  payload: string;
+}
```
Comment (steadyGuy) [inline frontend/src/redux/types/user.ts]: lot of eslint comments. hope it's gonna remove.

**Review by steadyGuy — APPROVED** [review]

File: frontend/package.json (hunk 2 above)
Comment (Yojuks) [reply frontend/package.json:13]: we didn't

File: frontend/src/redux/actions-creators/todo.ts (hunk 3 above)
Comment (Yojuks) [reply frontend/src/redux/actions-creators/todo.ts]: done

File: frontend/src/redux/actions-creators/user.ts (hunk 4 above)
Comment (Yojuks) [reply frontend/src/redux/actions-creators/user.ts]: maybe we need to discuss tomorrow

File: frontend/src/redux/components/TodoList.tsx (hunk 5 above)
Comment (Yojuks) [reply frontend/src/redux/components/TodoList.tsx]: it is only for example

**Review by romawht22 — APPROVED** [review]

File: frontend/package.json (hunk 1 above)
Comment (Yojuks) [reply frontend/package.json:24]: done

File: frontend/src/redux/components/TodoList.tsx (hunk 5 above)
Comment (Yojuks) [reply frontend/src/redux/components/TodoList.tsx]: fixed

**Pull request merged**